Daml Finance: Upgrade Tutorial (#533)
digital-asset/docs.daml.com: docs/2.8.0/docs/daml-finance/tutorials/upgrade/intro.rst

.. Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
.. SPDX-License-Identifier: Apache-2.0

Upgrade tutorials
#################

This section delves into the processes for upgrading existing contract instances on a Daml ledger.
A thorough understanding of the different types of upgrades is essential to choose the most suitable
method for various scenarios.

Upgrades are broadly classified into two categories:

* *Voluntary Upgrades:*

>   These upgrades only happen if you act, akin to an airline offering an upgrade from economy to

Reviewer: ```suggestion
  These upgrades only happen if the holding owner acts, akin to an airline offering an upgrade from economy to
```

Reviewer: When I first read this I was a bit unsure, until I realized what _you_ was referring to.

Reviewer 2: I modified it to make it more clear.

  business class. The upgrade materializes only if you opt-in and make the necessary payment.

* *Mandatory Upgrades:*

>   These upgrades happen inevitably, regardless of your preference. A common example is the automatic

Reviewer: ```suggestion
  These upgrades happen inevitably, regardless of the preference of the holding owner. A common example is the automatic
```
